lib/Backend/GlobOpt.cpp

```diff
@@ -13083,6 +13084,61 @@ GlobOpt::OptArraySrc(IR::Instr ** const instrRef, Value ** src1Val, Value ** src
     arraySrcOpt.Optimize();
 }
 
+void
+GlobOpt::OptArgLenAndConst(IR::Instr* instr, Value** src1Val)
+{
+    if (instr->m_func->IsStackArgsEnabled() && instr->IsInlined())
+    {
+        IR::Opnd* src1 = instr->GetSrc1();
+        auto replaceInstr = [&](IR::Instr* instr, IR::Opnd* newopnd, Value** src1Val)
+        {
+            this->CaptureByteCodeSymUses(instr);
+            instr->m_opcode = Js::OpCode::Ld_A;
+            instr->ReplaceSrc1(newopnd);
+            if (instr->HasBailOutInfo())
+            {
+                instr->ClearBailOutInfo();
+            }
+            *src1Val = this->OptSrc(instr->GetSrc1(), &instr);
+            instr->m_func->hasArgLenAndConstOpt = true;
+        };
+        switch(instr->m_opcode)
+        {
+            case Js::OpCode::LdLen_A:
+            {
+                if (CurrentBlockData()->IsArgumentsOpnd(src1))
+                {
+                    IR::AddrOpnd* newopnd = IR::AddrOpnd::New(Js::TaggedInt::ToVarUnchecked(instr->m_func->actualCount - 1), IR::AddrOpndKindConstantVar, instr->m_func);
+                    replaceInstr(instr, newopnd, src1Val);
```

Comment:
>replaceInstr [](start = 20, length = 12)

For LdLen_A, does the Ld_A eventually get type specialized to Ld_I4 (int 32 type spec)?

Reply:
Yes